### ISSUE TITLE
Map Extras don't spawn despite being marked by the autonote

### ISSUE THREAD

**Issue by gettingusedto** [CleverRaven/Cataclysm-DDA#67182]
### Describe the bug

For whatever reason, the map extras does not spawn into the world despite it being marked so by the autonote.

### Attach save file

[Co-Operative-trimmed.tar.gz](https://github.com/CleverRaven/Cataclysm-DDA/files/12170171/Co-Operative-trimmed.tar.gz)

### Steps to reproduce

1. either make a new world or use the save attached
2. go into a nearest map extra marked by the autonote (assuming you have the autonote turned on)
3. see that it doesnt appears to be there

### Expected behavior

Map extras spawned into the world.

### Screenshots

![Screenshot_20230726-160223](https://github.com/CleverRaven/Cataclysm-DDA/assets/78019001/f2eda3c7-9584-4f51-8fda-5ce669bef251)
![Screenshot_20230726-160235](https://github.com/CleverRaven/Cataclysm-DDA/assets/78019001/25b5419f-6470-4ca0-9dde-696249c6e174)

### Versions and configuration

- OS: Android
    - OS Version: Manufacturer: samsung; Model: SM-A710F; Release: 7.0; Incremental: A710FXXU2CRC2; 
- Game Version: 18b2783 [32-bit]
- Graphics Version: Tiles
- Game Language: English [en]
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food],
    Slowdown Fungal Growth [no_fungal_growth],
    Portal Storms Ignore NPCs [personal_portal_storms]
]

### Additional context

somehow the roadwork map extra can spawn in.

**Comment by waxcatape** [comment]
It's weird. Some spawn partially. Like corpse specials spawn grackens, krecks and molds but not actual corpses. Spiders spawn in without webs

**Comment by gettingusedto** [comment]
@Procyonae told me in the devcord that the nested placement got bugged into not spawning and as a result, any map extra that used nests don't spawn completely.

**Comment by andrei8l** [comment]
Try this
```diff
diff --git a/src/mapgen.cpp b/src/mapgen.cpp
index c125aeeaad..0d48d6a5e4 100644
--- a/src/mapgen.cpp
+++ b/src/mapgen.cpp
@@ -3608,7 +3608,7 @@ class jmapgen_nested : public jmapgen_piece
                             }
                         }
                     }
-                    return false;
+                    return neighbors.empty();
                 }
         };
 
```

**Comment by RenechCDDA** [comment]
Resolved by #67197
